File: sentiment-analysis/microsoft-cognitive-services-text-analytics/MicrosoftCognitiveServicesTextAnalyticsSentimentAnalysis.java
// For English, "language" is set to "en".

class Document {
    public String id, language, text;

    public Document(String id, String language, String text) {
        this.id = id;
        this.language = language;
        this.text = text;
    }
}

class Documents {
    public List<Document> documents;

    public Documents() {
        this.documents = new ArrayList<>();
    }
    public void add(String id, String language, String text) {
        this.documents.add (new Document (id, language, text));
    }
}

Documents documents = new Documents();
documents.add(id, language, textId);

Gson gson = new Gson();
String body = gson.toJson(documents);

String stringResponse = Unirest.post(host.concat(path))
    .header("Content-Type", "application/json")
    .header("Ocp-Apim-Subscription-Key", subscriptionKey)
    .body(body)
    .asString()
    .getBody();

Response response = gson.fromJson(stringResponse, Response.class);

File: sentiment-analysis/microsoft-cognitive-services-text-analytics/MicrosoftCognitiveServicesTextAnalyticsSentimentAnalysis.kt
// For English, set "language" to "en".

data class Document(val id: String, val language: String, var text: String)

data class SentimentAnalysis(val id: String, val score: Double)

data class SentimentAnalysisResponse(val documents: List<SentimentAnalysis>)

val host = "https://westeurope.api.cognitive.microsoft.com"
val path = "/text/analytics/v2.0/sentiment"
val url = host + path

val documents = listOf(Document("1", language, text))

val gson = Gson()
val body = gson.toJson(documents)

val stringResponse = Unirest.post(url)
    .header("Content-Type", "application/json")
    .header("Ocp-Apim-Subscription-Key", subscriptionKey)
    .body(body)
    .asString()
    .body

val response = gson.fromJson(stringResponse, SentimentAnalysisResponse::class.java)
val score = response.documents.single().score
